handlers/deltaValidationHandler.go

```diff
@@ -0,0 +1,50 @@
+package handlers
+
+import (
+	"fmt"
+	"github.com/companieshouse/chs-delta-api/config"
+	"github.com/companieshouse/chs-delta-api/helpers"
+	"github.com/companieshouse/chs-delta-api/validation"
+	"github.com/companieshouse/chs.go/log"
+	"net/http"
+)
+
+// DeltaValidationHandler offers a handler by which to publish an office-delta onto the officer-delta kafka topic.
+type DeltaValidationHandler struct {
+	h    helpers.Helper
+	chv  validation.CHValidator
+	cfg  *config.Config
+}
+
+// NewDeltaValidationHandler returns an DeltaValidatorHandler.
+func NewDeltaValidationHandler(h helpers.Helper, chv validation.CHValidator, cfg *config.Config) *DeltaValidationHandler {
+	return &DeltaValidationHandler{h: h, chv: chv, cfg: cfg}
+}
+
+// ServeHTTP accepts an incoming OfficerDelta request via a POST method, validates it
+// and then passes it to a Kafka service for further processing along with an officers-delta topic. If errors are
+// encountered then they will be returned via the ResponseWriter.
+func (kp *DeltaValidationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	contextId := kp.h.GetRequestIdFromHeader(r)
+
+	log.InfoC(contextId, fmt.Sprintf("Using the open api spec: "), log.Data{config.OpenApiSpecKey: kp.cfg.OpenApiSpec})
+
+	// Validate against the open API 3 spec before progressing any further.
+	errValidation, err := kp.chv.ValidateRequestAgainstOpenApiSpec(r, kp.cfg.OpenApiSpec, contextId)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.ErrorC(contextId, err, log.Data{config.MessageKey: "error occurred while trying to validate request"})
+		return
+	} else if errValidation != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		_, err = w.Write(errValidation)
+		if err != nil {
+			log.ErrorC(contextId, err, log.Data{config.MessageKey: "error occurred while trying to write response"})
+		}
+
+		return
+	}
```

Comment:
May be add this code as a member function of the validation struct and call it from validation handler as well as from delta handler?

Reply:
What specific code are we reusing here? Maybe as a line-number range?

Reply:
From line 31-49 will be exactly the same in all handlers @ch-tstrickland

Reply:
Lines 33 to 47. The call to kp.chv.ValidateRequestAgainstOpenApiSpec and the error handling after

Reply:
Perhaps this may be fine as-is. I wonder if the real effort to expend would be to make this all work off dynamic config, so that we only need to add config to add a new delta-type, and they all share the same handler.
I am ambivalent. I can see a good design that keeps the handlers all separate. I can see a good design that uses dynamic config and has one handler.